agent/cache/cache.go

```diff
@@ -499,7 +512,12 @@ func (c *Cache) fetch(key string, r getOptions, allowNew bool, attempt uint, ign
 				Index: entry.Index,
 			}
 		}
-
+		// TODO(context): handle the error without a panic when context is changed to one that is cancellable
+		if err := entry.FetchRateLimiter.Wait(context.Background()); err != nil {
```

Comment:
We have `lib.StopChannelContext{StopCh: shutdownCh}` that should be used here to cancel when shutting down. I think this is the least we should be doing here. I don't think the cache has access to the agent shutdown channel already though.

Reply:
Fixed in 7b23877ec74d40aca69941f890188ba4ae11dc91